### ISSUE TITLE
DataLoader worker (pid(s) 18056, 20540, 4512) exited unexpectedly

### ISSUE THREAD

**Issue by Aidenfaustine** [pytorch/pytorch#59451]
I'm new to pytorch. when i run commands in Jupyternotebook （windows  systems） %run main.py --task wilder --emo_dim arousal --feature_set egemaps --normalize --norm_opts y --d_rnn 128 --rnn_n_layers 4 --epochs 100 --batch_size 1024 --n_seeds 10 --win_len 200 --hop_len 100 --use_gpu --cache --save --save_path preds

I just want to reproduce the results:https://github.com/lstappen/MuSe2021

it reported error as follows:
DataLoader worker (pid(s) 18056, 20540, 4512) exited unexpectedly
Thanks
best wishes.

cc @SsnL @VitalyFedyunin @ejguan

**Comment by agolynski** [comment]
Could you add more details to have the full repro?

**Comment by ejguan** [comment]
Please don't open duplicate issue (#59438)
IIRC, Jupyter Notebook doesn't support python multiprocessing well (especially on windows). https://docs.python.org/3.8/library/multiprocessing.html#using-a-pool-of-workers
When you set `num_workers` > 0, DataLoader would start multiprocessing.
I suggest you directly run your script `main.py` with a normal python interpreter rather than the interactive interpreter (notebook)

**Comment by huangjin520** [comment]
> I'm new to pytorch. when i run commands in Jupyternotebook （windows systems） %run main.py --task wilder --emo_dim arousal --feature_set egemaps --normalize --norm_opts y --d_rnn 128 --rnn_n_layers 4 --epochs 100 --batch_size 1024 --n_seeds 10 --win_len 200 --hop_len 100 --use_gpu --cache --save --save_path preds
> 
> I just want to reproduce the results:https://github.com/lstappen/MuSe2021
> 
> it reported error as follows:
> DataLoader worker (pid(s) 18056, 20540, 4512) exited unexpectedly
> Thanks
> best wishes.
> 
> cc @SsnL @VitalyFedyunin @ejguan

I encounter the same question as you.It takes me 2hours to  search for solutions  , but i still don't find the answer. Do you find solution to solve it? And please tell me the question ,i need  your help.

**Comment by ejguan** [comment]
@huangjin520 Try to run your code out of notebook, or set num_workers to 0 for DataLoader in the jupyter notebook.

**Comment by huangjin520** [comment]
> @huangjin520 Try to run your code out of notebook, or set num_workers to 0 for DataLoader in the jupyter notebook.

Thanks for your help ! Now i run my script with pycharm and cmd, but it don't work for me. Because of my large image dataset , I need to  set num_workers >0 to Speed up training process.  So i want to know where is wrong, and how to solve it.

**Comment by ejguan** [comment]
> Thanks for your help ! Now i run my script with pycharm and cmd, but it don't work for me. Because of my large image dataset , I need to set num_workers >0 to Speed up training process. So i want to know where is wrong, and how to solve it.

If you can run your code with normal python interpreter in your terminal. You should be able to run with num_workers>0.

**Comment by huangjin520** [comment]
> > Thanks for your help ! Now i run my script with pycharm and cmd, but it don't work for me. Because of my large image dataset , I need to set num_workers >0 to Speed up training process. So i want to know where is wrong, and how to solve it.
> 
> If you can run your code with normal python interpreter in your terminal. You should be able to run with num_workers>0.
Thank you sincerely ! 
When i  set num_worker = 0, the script is interupted.
When i set num_worker >0, the script raise wrong information .

**Comment by ejguan** [comment]
@huangjin520 
Can you please open a new issue and share more detailed information about the Error when you set `num_worker=0`?

**Comment by huangjin520** [comment]
> > > Thanks for your help ! Now i run my script with pycharm and cmd, but it don't work for me. Because of my large image dataset , I need to set num_workers >0 to Speed up training process. So i want to know where is wrong, and how to solve it.
> > 
> > 
> > If you can run your code with normal python interpreter in your terminal. You should be able to run with num_workers>0.
> > Thank you sincerely !
> > When i  set num_worker = 0, the script is interupted.
> > When i set num_worker >0, the script raise wrong information .

According to these situations,  whether i can get this conclusion that the python interpreter or pytorch was installed wrong or  not .

**Comment by huangjin520** [comment]
> @huangjin520 Try to run your code out of notebook, or set num_workers to 0 for DataLoader in the jupyter notebook.

Thank you very very much! According to your several approaches, i  tried them one by one ,but the script still can't work. Especially , when i set num_workers = 0, the script is interrupted with raising the 'The kernel is dead'.

**Comment by huangjin520** [comment]
> > @huangjin520 Try to run your code out of notebook, or set num_workers to 0 for DataLoader in the jupyter notebook.
> 
> Thank you very very much! According to your several approaches, i tried them one by one ,but the script still can't work. Especially , when i set num_workers = 0, the script is interrupted with raising the 'The kernel is dead'.
I will open a new issue to show the details of my platform and script, and i hope you can give me more solutions to overcome the bug.

**Comment by huangjin520** [comment]
> @huangjin520
> Can you please open a new issue and share more detailed information about the Error when you set `num_worker=0`?

 here is the details of this question. I'm sincerely grateful to get your help and instruction.[https://github.com/pytorch/pytorch/issues/62878](url)

**Comment by Aidenfaustine** [comment]
很抱歉，google邮箱不是经常看。
你可以联系微信V：shai265130.

好运

**Comment by 18325614754** [comment]
Have you solved this problem now? I have also encountered the same problem now, looking forward to your reply!
